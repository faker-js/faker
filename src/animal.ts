import type { Faker } from '.';

export class Animal {
  constructor(private readonly faker: Faker) {
    // Bind `this` so namespaced is working correctly
    for (const name of Object.getOwnPropertyNames(Animal.prototype)) {
      if (name === 'constructor' || typeof this[name] !== 'function') {
        continue;
      }
      this[name] = this[name].bind(this);
    }
  }

  /**
   * Returns a random dog breed.
   *
   */
  dog(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.dog);
  }

  /**
   * Returns a random cat breed.
   *
   */
  cat(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.cat);
  }

  /**
   * Returns a random snake species.
   *
   */
  snake(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.snake);
  }

  /**
   * Returns a random bear species.
   *
   */
  bear(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.bear);
  }

  /**
   * Returns a random lion species.
   *
   */
  lion(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.lion);
  }

  /**
   * Returns a random cetacean species.
   *
   * @method faker.animal.cetacean
   */
  cetacean(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.animal.cetacean
    );
  }

  /**
   * Returns a random horse breed.
   *
   */
  horse(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.horse);
  }

  /**
   * Returns a random bird species.
   *
   */
  bird(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.bird);
  }

  /**
   * Returns a random cow species.
   *
   */
  cow(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.cow);
  }

  /**
   * Returns a random fish species.
   *
   */
  fish(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.fish);
  }

  /**
   * Returns a random crocodilian species.
   *
   */
  crocodilia(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.animal.crocodilia
    );
  }

  /**
   * Returns a random insect species.
   *
   */
  insect(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.insect);
  }

  /**
   * Returns a random rabbit species
   *
   */
  rabbit(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.rabbit);
  }

  /**
   * Returns a random animal type.
   *
   */
  type(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.type);
  }
}
