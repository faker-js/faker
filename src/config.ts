/**
 * The possible configuration options, that can be set.
 * This type exists to be extended for plugins via type augmentation.
 *
 * The `@default` tag is used to indicate the default value, that should be used if, the config is absent.
 */
export interface FakerConfig {
  /**
   * The function used to generate the `refDate` date instance, if not provided as method param.
   * The function must return a new valid `Date` instance for every call.
   *
   * @see [Reproducible Results](https://fakerjs.dev/guide/usage.html#reproducible-results)
   * @see faker.seed(): For generating reproducible values.
   *
   * @since 9.0.0
   *
   * @default () => new Date()
   */
  defaultRefDate?: () => Date;
}
