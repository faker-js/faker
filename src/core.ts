import type { FakerConfig } from './config';
import type { LocaleDefinition } from './definitions';
import type { Randomizer } from './randomizer';

/**
 * The core grants access to the locale data, the randomizer and config settings.
 */
export interface FakerCore {
  /**
   * The locale data associated with this instance.
   *
   * Always present, but it might be empty if the locale data is not available.
   */
  readonly locale: LocaleDefinition;

  /**
   * The randomizer used to generate random values.
   */
  readonly randomizer: Randomizer;

  /**
   * The configuration settings used by this instance.
   */
  readonly config: FakerConfig;
}
