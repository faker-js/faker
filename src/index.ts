import { Faker } from './faker';
import allLocales from './locales';

export type {
  AnimalDefinitions,
  ColorDefinitions,
  CommerceDefinitions,
  CommerceProductNameDefinitions,
  CompanyDefinitions,
  DatabaseDefinitions,
  DateDefinitions,
  DateEntryDefinition,
  FinanceCurrencyEntryDefinitions,
  FinanceDefinitions,
  HackerDefinitions,
  InternetDefinitions,
  LocaleDefinition,
  /** @deprecated Use LocationDefinitions instead */
  LocationDefinitions as AddressDefinitions,
  LocationDefinitions,
  LoremDefinitions,
  MusicDefinitions,
  /** @deprecated Use PersonDefinitions instead */
  PersonDefinitions as NameDefinitions,
  PersonDefinitions,
  /** @deprecated Use PersonTitleDefinitions instead */
  PersonTitleDefinitions as NameTitleDefinitions,
  PersonTitleDefinitions,
  PhoneNumberDefinitions,
  ScienceDefinitions,
  SystemDefinitions,
  SystemMimeTypeEntryDefinitions,
  VehicleDefinitions,
  WordDefinitions,
} from './definitions';
export { FakerError } from './errors/faker-error';
export type { FakerOptions, UsableLocale, UsedLocales } from './faker';
export type { AnimalModule } from './modules/animal';
export type {
  Casing,
  ColorFormat,
  ColorModule,
  CSSFunction,
  CSSSpace,
  NumberColorFormat,
  StringColorFormat,
} from './modules/color';
export type { CommerceModule } from './modules/commerce';
export type { CompanyModule } from './modules/company';
export type { DatabaseModule } from './modules/database';
export type { DatatypeModule } from './modules/datatype';
export type { DateModule } from './modules/date';
export type { FinanceModule } from './modules/finance';
export type { GitModule } from './modules/git';
export type { HackerModule } from './modules/hacker';
export type { HelpersModule } from './modules/helpers';
export type { ImageModule } from './modules/image';
export type { InternetModule } from './modules/internet';
export type {
  /** @deprecated Use LocationModule instead */
  LocationModule as AddressModule,
  LocationModule,
} from './modules/location';
export type { LoremModule } from './modules/lorem';
export type { MusicModule } from './modules/music';
export { Sex } from './modules/person';
export type {
  /** @deprecated Use PersonModule instead */
  PersonModule as NameModule,
  PersonModule,
  SexType,
} from './modules/person';
export type { PhoneModule } from './modules/phone';
export type { RandomModule } from './modules/random';
export type { ChemicalElement, ScienceModule, Unit } from './modules/science';
export type { StringModule } from './modules/string';
export type { SystemModule } from './modules/system';
export type { VehicleModule } from './modules/vehicle';
export type { WordModule } from './modules/word';
export { Faker };

// since we are requiring the top level of faker, load all locales by default
export const faker: Faker = new Faker({
  locales: allLocales,
});
