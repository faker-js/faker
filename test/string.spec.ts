import { describe, expect, it } from 'vitest';
import { faker, FakerError } from '../src';
import { seededTests } from './support/seededRuns';
import { times } from './support/times';

const NON_SEEDED_BASED_RUN = 5;

describe('string', () => {
  seededTests(faker, 'string', (t) => {
    t.describe('alpha', (t) => {
      t.it('noArgs')
        .itRepeated('with length parameter', 5, 5)
        .it('with length', { length: 6 })
        .it('with casing = lower', { casing: 'lower' })
        .it('with casing = upper', { casing: 'upper' })
        .it('with casing = mixed', { casing: 'mixed' })
        .it('with bannedChars', { bannedChars: 'abcdefghijk' })
        .it('with length, casing and bannedChars', {
          length: 7,
          casing: 'lower',
          bannedChars: 'lmnopqrstu',
        });
    });

    t.describe('alphanumeric', (t) => {
      t.it('noArgs')
        .itRepeated('with length parameter', 5, 5)
        .it('with length', { length: 6 })
        .it('with casing = lower', { casing: 'lower' })
        .it('with casing = upper', { casing: 'upper' })
        .it('with casing = mixed', { casing: 'mixed' })
        .it('with bannedChars', { bannedChars: 'abcdefghijk12345' })
        .it('with length, casing and bannedChars', {
          length: 7,
          casing: 'lower',
          bannedChars: 'lmnopqrstu67890',
        });
    });

    t.describe('hexadecimal', (t) => {
      t.it('noArgs')
        .it('with length', { length: 6 })
        .it('with casing = lower', { casing: 'lower' })
        .it('with casing = upper', { casing: 'upper' })
        .it('with casing = mixed', { casing: 'mixed' })
        .it('with custom prefix', { prefix: 'hex_' })
        .it('with length, casing and empty prefix', {
          length: 7,
          casing: 'lower',
          prefix: '',
        });
    });

    t.describe('numeric', (t) => {
      t.it('noArgs')
        .itRepeated('with length parameter', 5, 5)
        .it('with length', { length: 6 })
        .it('with allowLeadingZeros', { allowLeadingZeros: true })
        .it('with bannedDigits', { bannedDigits: '12345' })
        .it('with length, allowLeadingZeros and bannedDigits', {
          length: 7,
          allowLeadingZeros: true,
          bannedDigits: '12345',
        });
    });

    t.describe('sample', (t) => {
      t.it('noArgs').itRepeated('with length parameter', 5, 5);
    });

    t.itRepeated('uuid', 5);
  });

  describe(`random seeded tests for seed ${faker.seed()}`, () => {
    for (let i = 1; i <= NON_SEEDED_BASED_RUN; i++) {
      describe('alpha', () => {
        it('should return single letter when no length provided', () => {
          const actual = faker.string.alpha();

          expect(actual).toHaveLength(1);
        });

        it('should return any letters when no option is provided', () => {
          const actual = faker.string.alpha();

          expect(actual).toMatch(/^[a-zA-Z]$/);
        });

        it.each([
          ['upper', /^[A-Z]{250}$/],
          ['lower', /^[a-z]{250}$/],
          ['mixed', /^[a-zA-Z]{250}$/],
        ] as const)('should return %s-case', (casing, pattern) => {
          const actual = faker.string.alpha({ length: 250, casing });
          expect(actual).toMatch(pattern);
        });

        it('should generate many random letters', () => {
          const actual = faker.string.alpha(5);

          expect(actual).toHaveLength(5);
        });

        it.each([0, -1, -100])(
          'should return empty string when length is <= 0',
          (length) => {
            const actual = faker.string.alpha(length);

            expect(actual).toBe('');
          }
        );

        it('should be able to ban some characters', () => {
          const actual = faker.string.alpha({
            length: 5,
            casing: 'lower',
            bannedChars: ['a', 'p'],
          });

          expect(actual).toHaveLength(5);
          expect(actual).toMatch(/^[b-oq-z]{5}$/);
        });

        it('should be able to ban some characters via string', () => {
          const actual = faker.string.alpha({
            length: 5,
            casing: 'lower',
            bannedChars: 'ap',
          });

          expect(actual).toHaveLength(5);
          expect(actual).toMatch(/^[b-oq-z]{5}$/);
        });

        it('should be able handle mistake in banned characters array', () => {
          const alphaText = faker.string.alpha({
            length: 5,
            casing: 'lower',
            bannedChars: ['a', 'a', 'p'],
          });

          expect(alphaText).toHaveLength(5);
          expect(alphaText).toMatch(/^[b-oq-z]{5}$/);
        });

        it('should throw if all possible characters being banned', () => {
          const bannedChars = 'abcdefghijklmnopqrstuvwxyz'.split('');
          expect(() =>
            faker.string.alpha({
              length: 5,
              casing: 'lower',
              bannedChars,
            })
          ).toThrowError(
            new FakerError(
              'Unable to generate string, because all possible characters are banned.'
            )
          );
        });

        it('should not mutate the input object', () => {
          const input: {
            length: number;
            casing: 'mixed';
            bannedChars: string[];
          } = Object.freeze({
            length: 5,
            casing: 'mixed',
            bannedChars: ['a', '%'],
          });

          expect(() => faker.string.alpha(input)).not.toThrow();
          expect(input.bannedChars).toEqual(['a', '%']);
        });
      });

      describe('alphaNumeric', () => {
        it('should generate single character when no additional argument was provided', () => {
          const actual = faker.string.alphanumeric();

          expect(actual).toHaveLength(1);
        });

        it.each([
          ['upper', /^[A-Z0-9]{250}$/],
          ['lower', /^[a-z0-9]{250}$/],
          ['mixed', /^[a-zA-Z0-9]{250}$/],
        ] as const)('should return %s-case', (casing, pattern) => {
          const actual = faker.string.alphanumeric({ length: 250, casing });
          expect(actual).toMatch(pattern);
        });

        it('should generate many random characters', () => {
          const actual = faker.string.alphanumeric(5);

          expect(actual).toHaveLength(5);
        });

        it.each([0, -1, -100])(
          'should return empty string when length is <= 0',
          (length) => {
            const actual = faker.string.alphanumeric(length);

            expect(actual).toBe('');
          }
        );

        it('should be able to ban all alphabetic characters', () => {
          const bannedChars = 'abcdefghijklmnopqrstuvwxyz'.split('');
          const alphaText = faker.string.alphanumeric({
            length: 5,
            casing: 'lower',
            bannedChars,
          });

          expect(alphaText).toHaveLength(5);
          for (const bannedChar of bannedChars) {
            expect(alphaText).not.includes(bannedChar);
          }
        });

        it('should be able to ban all alphabetic characters via string', () => {
          const bannedChars = 'abcdefghijklmnopqrstuvwxyz';
          const alphaText = faker.string.alphanumeric({
            length: 5,
            casing: 'lower',
            bannedChars,
          });

          expect(alphaText).toHaveLength(5);
          for (const bannedChar of bannedChars) {
            expect(alphaText).not.includes(bannedChar);
          }
        });

        it('should be able to ban all numeric characters', () => {
          const bannedChars = '0123456789'.split('');
          const alphaText = faker.string.alphanumeric({
            length: 5,
            bannedChars,
          });

          expect(alphaText).toHaveLength(5);
          for (const bannedChar of bannedChars) {
            expect(alphaText).not.includes(bannedChar);
          }
        });

        it('should be able to ban all numeric characters via string', () => {
          const bannedChars = '0123456789';
          const alphaText = faker.string.alphanumeric({
            length: 5,
            bannedChars,
          });

          expect(alphaText).toHaveLength(5);
          for (const bannedChar of bannedChars) {
            expect(alphaText).not.includes(bannedChar);
          }
        });

        it('should be able to handle mistake in banned characters array', () => {
          const alphaText = faker.string.alphanumeric({
            length: 5,
            casing: 'lower',
            bannedChars: ['a', 'p', 'a'],
          });

          expect(alphaText).toHaveLength(5);
          expect(alphaText).toMatch(/^[0-9b-oq-z]{5}$/);
        });

        it('should throw if all possible characters being banned', () => {
          const bannedChars = 'abcdefghijklmnopqrstuvwxyz0123456789'.split('');
          expect(() =>
            faker.string.alphanumeric({
              length: 5,
              casing: 'lower',
              bannedChars,
            })
          ).toThrowError(
            new FakerError(
              'Unable to generate string, because all possible characters are banned.'
            )
          );
        });

        it('should throw if all possible characters being banned via string', () => {
          const bannedChars = 'abcdefghijklmnopqrstuvwxyz0123456789';
          expect(() =>
            faker.string.alphanumeric({
              length: 5,
              casing: 'lower',
              bannedChars,
            })
          ).toThrowError();
        });

        it('should not mutate the input object', () => {
          const input: {
            length: number;
            bannedChars: string[];
          } = Object.freeze({
            length: 5,
            bannedChars: ['a', '0', '%'],
          });

          expect(() => faker.string.alphanumeric(input)).not.toThrow();
          expect(input.bannedChars).toEqual(['a', '0', '%']);
        });
      });

      describe(`hexadecimal`, () => {
        it('generates single hex character when no additional argument was provided', () => {
          const hex = faker.string.hexadecimal();
          expect(hex).toMatch(/^0x[0-9a-f]*$/i);
          expect(hex).toHaveLength(3);
        });

        it('generates a random hex string', () => {
          const hex = faker.string.hexadecimal({
            length: 5,
            prefix: '',
          });
          expect(hex).toMatch(/^[0-9a-f]*$/i);
          expect(hex).toHaveLength(5);
        });
      });

      describe('numeric', () => {
        it('should return single digit when no length provided', () => {
          const actual = faker.string.numeric();

          expect(actual).toHaveLength(1);
          expect(actual).toMatch(/^[1-9]$/);
        });

        it.each(times(100))(
          'should generate random value with a length of %s',
          (length) => {
            const actual = faker.string.numeric(length);

            expect(actual).toHaveLength(length);
            expect(actual).toMatch(/^[1-9][0-9]*$/);
          }
        );

        it('should return empty string with a length of 0', () => {
          const actual = faker.string.numeric(0);

          expect(actual).toHaveLength(0);
        });

        it('should return empty string with a negative length', () => {
          const actual = faker.string.numeric(-10);

          expect(actual).toHaveLength(0);
        });

        it('should return a valid numeric string with provided length', () => {
          const actual = faker.string.numeric(1000);

          expect(actual).toBeTypeOf('string');
          expect(actual).toHaveLength(1000);
          expect(actual).toMatch(/^[1-9][0-9]+$/);
        });

        it('should allow leading zeros via option', () => {
          const actual = faker.string.numeric({
            length: 15,
            allowLeadingZeros: true,
          });

          expect(actual).toMatch(/^[0-9]+$/);
        });

        it('should allow leading zeros via option and all other digits banned', () => {
          const actual = faker.string.numeric({
            length: 4,
            allowLeadingZeros: true,
            bannedDigits: '123456789'.split(''),
          });

          expect(actual).toBe('0000');
        });

        it('should allow leading zeros via option and all other digits banned via string', () => {
          const actual = faker.string.numeric({
            length: 4,
            allowLeadingZeros: true,
            bannedDigits: '123456789',
          });

          expect(actual).toBe('0000');
        });

        it('should fail on leading zeros via option and all other digits banned', () => {
          expect(() =>
            faker.string.numeric({
              length: 4,
              allowLeadingZeros: false,
              bannedDigits: '123456789'.split(''),
            })
          ).toThrowError(
            new FakerError(
              'Unable to generate numeric string, because all possible digits are banned.'
            )
          );
        });

        it('should fail on leading zeros via option and all other digits banned via string', () => {
          expect(() =>
            faker.string.numeric({
              length: 4,
              allowLeadingZeros: false,
              bannedDigits: '123456789',
            })
          ).toThrowError(
            new FakerError(
              'Unable to generate numeric string, because all possible digits are banned.'
            )
          );
        });

        it('should ban all digits passed via bannedDigits', () => {
          const actual = faker.string.numeric({
            length: 1000,
            bannedDigits: 'c84U1'.split(''),
          });

          expect(actual).toHaveLength(1000);
          expect(actual).toMatch(/^[0235679]{1000}$/);
        });

        it('should ban all digits passed via bannedDigits via string', () => {
          const actual = faker.string.numeric({
            length: 1000,
            bannedDigits: 'c84U1',
          });

          expect(actual).toHaveLength(1000);
          expect(actual).toMatch(/^[0235679]{1000}$/);
        });
      });

      describe('sample', () => {
        it('should generate a string value', () => {
          const generatedString = faker.string.sample();
          expect(generatedString).toBeTypeOf('string');
          expect(generatedString).toHaveLength(10);
        });

        it('should return empty string if negative length is passed', () => {
          const negativeValue = faker.datatype.number({ min: -1000, max: -1 });
          const generatedString = faker.string.sample(negativeValue);
          expect(generatedString).toBe('');
          expect(generatedString).toHaveLength(0);
        });

        it('should return string with length of 2^20 if bigger length value is passed', () => {
          const overMaxValue = Math.pow(2, 28);
          const generatedString = faker.string.sample(overMaxValue);
          expect(generatedString).toHaveLength(Math.pow(2, 20));
        });

        it('should return string with a specific length', () => {
          const length = 1337;
          const generatedString = faker.string.sample(length);
          expect(generatedString).toHaveLength(length);
        });
      });

      describe(`uuid`, () => {
        it('generates a valid UUID', () => {
          const UUID = faker.string.uuid();
          const RFC4122 =
            /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/;
          expect(UUID).toMatch(RFC4122);
        });
      });
    }
  });
});
