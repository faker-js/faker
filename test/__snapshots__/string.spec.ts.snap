// Vitest Snapshot v1

exports[`string > 42 > alpha > noArgs 1`] = `"t"`;

exports[`string > 42 > alpha > with bannedChars 1`] = `"A"`;

exports[`string > 42 > alpha > with casing = lower 1`] = `"j"`;

exports[`string > 42 > alpha > with casing = mixed 1`] = `"t"`;

exports[`string > 42 > alpha > with casing = upper 1`] = `"J"`;

exports[`string > 42 > alpha > with length 1`] = `"tPXjMO"`;

exports[`string > 42 > alpha > with length parameter 1`] = `"tPXjM"`;

exports[`string > 42 > alpha > with length parameter 2`] = `"OFFix"`;

exports[`string > 42 > alpha > with length parameter 3`] = `"ifdxT"`;

exports[`string > 42 > alpha > with length parameter 4`] = `"rFhKH"`;

exports[`string > 42 > alpha > with length parameter 5`] = `"bcYLR"`;

exports[`string > 42 > alpha > with length, casing and bannedChars 1`] = `"fwzcvwj"`;

exports[`string > 42 > alphanumeric > noArgs 1`] = `"n"`;

exports[`string > 42 > alphanumeric > with bannedChars 1`] = `"x"`;

exports[`string > 42 > alphanumeric > with casing = lower 1`] = `"d"`;

exports[`string > 42 > alphanumeric > with casing = mixed 1`] = `"n"`;

exports[`string > 42 > alphanumeric > with casing = upper 1`] = `"D"`;

exports[`string > 42 > alphanumeric > with length 1`] = `"nNWbJM"`;

exports[`string > 42 > alphanumeric > with length parameter 1`] = `"nNWbJ"`;

exports[`string > 42 > alphanumeric > with length parameter 2`] = `"MBB9r"`;

exports[`string > 42 > alphanumeric > with length parameter 3`] = `"963sR"`;

exports[`string > 42 > alphanumeric > with length parameter 4`] = `"kB8HE"`;

exports[`string > 42 > alphanumeric > with length parameter 5`] = `"13YIP"`;

exports[`string > 42 > alphanumeric > with length, casing and bannedChars 1`] = `"cvy4kvh"`;

exports[`string > 42 > hexadecimal > noArgs 1`] = `"0x8"`;

exports[`string > 42 > hexadecimal > with casing = lower 1`] = `"0x8"`;

exports[`string > 42 > hexadecimal > with casing = mixed 1`] = `"0x8"`;

exports[`string > 42 > hexadecimal > with casing = upper 1`] = `"0x8"`;

exports[`string > 42 > hexadecimal > with custom prefix 1`] = `"hex_8"`;

exports[`string > 42 > hexadecimal > with length 1`] = `"0x8BE4AB"`;

exports[`string > 42 > hexadecimal > with length, casing and empty prefix 1`] = `"8be4abd"`;

exports[`string > 42 > numeric > noArgs 1`] = `"4"`;

exports[`string > 42 > numeric > with allowLeadingZeros 1`] = `"3"`;

exports[`string > 42 > numeric > with bannedDigits 1`] = `"7"`;

exports[`string > 42 > numeric > with length 1`] = `"479177"`;

exports[`string > 42 > numeric > with length parameter 1`] = `"47917"`;

exports[`string > 42 > numeric > with length parameter 2`] = `"85514"`;

exports[`string > 42 > numeric > with length parameter 3`] = `"20048"`;

exports[`string > 42 > numeric > with length parameter 4`] = `"46176"`;

exports[`string > 42 > numeric > with length parameter 5`] = `"10978"`;

exports[`string > 42 > numeric > with length, allowLeadingZeros and bannedDigits 1`] = `"6890887"`;

exports[`string > 42 > sample > noArgs 1`] = `"Cky2eiXX/J"`;

exports[`string > 42 > sample > with length parameter 1`] = `"Cky2e"`;

exports[`string > 42 > sample > with length parameter 2`] = `"iXX/J"`;

exports[`string > 42 > sample > with length parameter 3`] = `"/*&Kq"`;

exports[`string > 42 > sample > with length parameter 4`] = `"@X.b]"`;

exports[`string > 42 > sample > with length parameter 5`] = `"\\"&{dn"`;

exports[`string > 42 > uuid 1`] = `"5cf2bc99-2721-407d-992b-a00fbdf302f2"`;

exports[`string > 42 > uuid 2`] = `"94980604-8962-404f-9371-c9368f970d9a"`;

exports[`string > 42 > uuid 3`] = `"2710fff9-c640-413a-b7a1-97d02e642ac4"`;

exports[`string > 42 > uuid 4`] = `"6838920f-dc7f-46ee-9be5-19380f5d6b48"`;

exports[`string > 42 > uuid 5`] = `"d95f4984-24c2-410f-ac63-400d3bbbcc91"`;

exports[`string > 1211 > alpha > noArgs 1`] = `"W"`;

exports[`string > 1211 > alpha > with bannedChars 1`] = `"X"`;

exports[`string > 1211 > alpha > with casing = lower 1`] = `"y"`;

exports[`string > 1211 > alpha > with casing = mixed 1`] = `"W"`;

exports[`string > 1211 > alpha > with casing = upper 1`] = `"Y"`;

exports[`string > 1211 > alpha > with length 1`] = `"WxUOlg"`;

exports[`string > 1211 > alpha > with length parameter 1`] = `"WxUOl"`;

exports[`string > 1211 > alpha > with length parameter 2`] = `"gZbIi"`;

exports[`string > 1211 > alpha > with length parameter 3`] = `"JkNvs"`;

exports[`string > 1211 > alpha > with length parameter 4`] = `"iKMQa"`;

exports[`string > 1211 > alpha > with length parameter 5`] = `"NIILR"`;

exports[`string > 1211 > alpha > with length, casing and bannedChars 1`] = `"yhywdcz"`;

exports[`string > 1211 > alphanumeric > noArgs 1`] = `"V"`;

exports[`string > 1211 > alphanumeric > with bannedChars 1`] = `"W"`;

exports[`string > 1211 > alphanumeric > with casing = lower 1`] = `"x"`;

exports[`string > 1211 > alphanumeric > with casing = mixed 1`] = `"V"`;

exports[`string > 1211 > alphanumeric > with casing = upper 1`] = `"X"`;

exports[`string > 1211 > alphanumeric > with length 1`] = `"VsTMd8"`;

exports[`string > 1211 > alphanumeric > with length parameter 1`] = `"VsTMd"`;

exports[`string > 1211 > alphanumeric > with length parameter 2`] = `"8Z2F9"`;

exports[`string > 1211 > alphanumeric > with length parameter 3`] = `"GdLql"`;

exports[`string > 1211 > alphanumeric > with length parameter 4`] = `"aHKO0"`;

exports[`string > 1211 > alphanumeric > with length parameter 5`] = `"LFEJP"`;

exports[`string > 1211 > alphanumeric > with length, casing and bannedChars 1`] = `"yexv53z"`;

exports[`string > 1211 > hexadecimal > noArgs 1`] = `"0xE"`;

exports[`string > 1211 > hexadecimal > with casing = lower 1`] = `"0xe"`;

exports[`string > 1211 > hexadecimal > with casing = mixed 1`] = `"0xE"`;

exports[`string > 1211 > hexadecimal > with casing = upper 1`] = `"0xE"`;

exports[`string > 1211 > hexadecimal > with custom prefix 1`] = `"hex_E"`;

exports[`string > 1211 > hexadecimal > with length 1`] = `"0xEaDB42"`;

exports[`string > 1211 > hexadecimal > with length, casing and empty prefix 1`] = `"eadb42f"`;

exports[`string > 1211 > numeric > noArgs 1`] = `"9"`;

exports[`string > 1211 > numeric > with allowLeadingZeros 1`] = `"9"`;

exports[`string > 1211 > numeric > with bannedDigits 1`] = `"9"`;

exports[`string > 1211 > numeric > with length 1`] = `"948721"`;

exports[`string > 1211 > numeric > with length parameter 1`] = `"94872"`;

exports[`string > 1211 > numeric > with length parameter 2`] = `"29061"`;

exports[`string > 1211 > numeric > with length parameter 3`] = `"72743"`;

exports[`string > 1211 > numeric > with length parameter 4`] = `"26780"`;

exports[`string > 1211 > numeric > with length parameter 5`] = `"76678"`;

exports[`string > 1211 > numeric > with length, allowLeadingZeros and bannedDigits 1`] = `"9798609"`;

exports[`string > 1211 > sample > noArgs 1`] = `"wKti5-}$_/"`;

exports[`string > 1211 > sample > with length parameter 1`] = `"wKti5"`;

exports[`string > 1211 > sample > with length parameter 2`] = `"-}$_/"`;

exports[`string > 1211 > sample > with length parameter 3`] = `"\`4hHA"`;

exports[`string > 1211 > sample > with length parameter 4`] = `"0afl\\""`;

exports[`string > 1211 > sample > with length parameter 5`] = `"h^]dn"`;

exports[`string > 1211 > uuid 1`] = `"e7ec32f0-a2a3-4c65-abbd-0caabde64dfd"`;

exports[`string > 1211 > uuid 2`] = `"f379e325-9f7c-4064-a086-f23942b68e5f"`;

exports[`string > 1211 > uuid 3`] = `"d4694649-2183-4b32-8bd7-a336639d6997"`;

exports[`string > 1211 > uuid 4`] = `"10ab829b-742c-4a8b-b732-98d718d77069"`;

exports[`string > 1211 > uuid 5`] = `"7b91ce88-effb-4d1d-93bb-ad759e00b86c"`;

exports[`string > 1337 > alpha > noArgs 1`] = `"n"`;

exports[`string > 1337 > alpha > with bannedChars 1`] = `"v"`;

exports[`string > 1337 > alpha > with casing = lower 1`] = `"g"`;

exports[`string > 1337 > alpha > with casing = mixed 1`] = `"n"`;

exports[`string > 1337 > alpha > with casing = upper 1`] = `"G"`;

exports[`string > 1337 > alpha > with length 1`] = `"nDiloC"`;

exports[`string > 1337 > alpha > with length parameter 1`] = `"nDilo"`;

exports[`string > 1337 > alpha > with length parameter 2`] = `"Cxbqm"`;

exports[`string > 1337 > alpha > with length parameter 3`] = `"AEnrY"`;

exports[`string > 1337 > alpha > with length parameter 4`] = `"oMpfP"`;

exports[`string > 1337 > alpha > with length parameter 5`] = `"ueGsg"`;

exports[`string > 1337 > alpha > with length, casing and bannedChars 1`] = `"eicdeih"`;

exports[`string > 1337 > alphanumeric > noArgs 1`] = `"g"`;

exports[`string > 1337 > alphanumeric > with bannedChars 1`] = `"s"`;

exports[`string > 1337 > alphanumeric > with casing = lower 1`] = `"9"`;

exports[`string > 1337 > alphanumeric > with casing = mixed 1`] = `"g"`;

exports[`string > 1337 > alphanumeric > with casing = upper 1`] = `"9"`;

exports[`string > 1337 > alphanumeric > with length 1`] = `"gy9dhx"`;

exports[`string > 1337 > alphanumeric > with length parameter 1`] = `"gy9dh"`;

exports[`string > 1337 > alphanumeric > with length parameter 2`] = `"xs2je"`;

exports[`string > 1337 > alphanumeric > with length parameter 3`] = `"wAgkY"`;

exports[`string > 1337 > alphanumeric > with length parameter 4`] = `"gJj7N"`;

exports[`string > 1337 > alphanumeric > with length parameter 5`] = `"n5Cm7"`;

exports[`string > 1337 > alphanumeric > with length, casing and bannedChars 1`] = `"ag45age"`;

exports[`string > 1337 > hexadecimal > noArgs 1`] = `"0x5"`;

exports[`string > 1337 > hexadecimal > with casing = lower 1`] = `"0x5"`;

exports[`string > 1337 > hexadecimal > with casing = mixed 1`] = `"0x5"`;

exports[`string > 1337 > hexadecimal > with casing = upper 1`] = `"0x5"`;

exports[`string > 1337 > hexadecimal > with custom prefix 1`] = `"hex_5"`;

exports[`string > 1337 > hexadecimal > with length 1`] = `"0x5c346b"`;

exports[`string > 1337 > hexadecimal > with length, casing and empty prefix 1`] = `"5c346ba"`;

exports[`string > 1337 > numeric > noArgs 1`] = `"3"`;

exports[`string > 1337 > numeric > with allowLeadingZeros 1`] = `"2"`;

exports[`string > 1337 > numeric > with bannedDigits 1`] = `"7"`;

exports[`string > 1337 > numeric > with length 1`] = `"351225"`;

exports[`string > 1337 > numeric > with length parameter 1`] = `"35122"`;

exports[`string > 1337 > numeric > with length parameter 2`] = `"54032"`;

exports[`string > 1337 > numeric > with length parameter 3`] = `"55239"`;

exports[`string > 1337 > numeric > with length parameter 4`] = `"37318"`;

exports[`string > 1337 > numeric > with length parameter 5`] = `"40631"`;

exports[`string > 1337 > numeric > with length, allowLeadingZeros and bannedDigits 1`] = `"6706677"`;

exports[`string > 1337 > sample > noArgs 1`] = `"9U/4:SK$>6"`;

exports[`string > 1337 > sample > with length parameter 1`] = `"9U/4:"`;

exports[`string > 1337 > sample > with length parameter 2`] = `"SK$>6"`;

exports[`string > 1337 > sample > with length parameter 3`] = `"QX9@{"`;

exports[`string > 1337 > sample > with length parameter 4`] = `":e=+k"`;

exports[`string > 1337 > sample > with length parameter 5`] = `"D)[B,"`;

exports[`string > 1337 > uuid 1`] = `"48234870-5389-445f-8b41-c61a52bf27dc"`;

exports[`string > 1337 > uuid 2`] = `"cc057669-8c53-474d-a677-226d3e8ed92f"`;

exports[`string > 1337 > uuid 3`] = `"fe6d8b8b-0db9-4fa2-b265-abc0a5d0ccf5"`;

exports[`string > 1337 > uuid 4`] = `"0b87afbd-8949-4dfb-8d04-19f4fe7458b2"`;

exports[`string > 1337 > uuid 5`] = `"0bcea83c-a7ea-428e-85db-d448f2b777a6"`;
